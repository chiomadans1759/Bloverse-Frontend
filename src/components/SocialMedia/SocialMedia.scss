@import 'sass-rem';

#bl-socail-media {
  &-title {
    font-size: rem(16px);
  }
  .social-icon {
    height: rem(14px);
    width: rem(14px);
  }
  .social-text {
    font-size: rem(12px);
  }
}
