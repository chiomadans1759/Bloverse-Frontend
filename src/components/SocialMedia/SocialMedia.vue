<template>
  <div id="bl-socail-media" class="my-4">
    <p class="mb-2 social-title" :class="title"><strong>Social Media</strong></p>
    <div class="d-flex flex-wrap">
      <div class="mr-2">
        <div class="border rounded p-2 mb-1 d-flex align-items-center">
          <img class="social-icon" :class="customSocialIconStyle"
            src="@/assets/facebook-square.svg"
            alt="facebook"
          />
          <h6 class="social-text text-center pl-2">Facebook</h6>
        </div>
      </div>
      <div class="mr-2">
        <div class="border rounded p-2 mb-1 d-flex align-items-center">
          <img class="social-icon" :class="customSocialIconStyle"
            src="@/assets/twitter-square.svg"
            alt="facebook"
          />
          <h6 class="social-text text-center pl-2">Twitter</h6>
        </div>
      </div>
      <div class="mr-2">
        <div class="border rounded p-2 mb-1 d-flex align-items-center">
          <img class="social-icon" :class="customSocialIconStyle"
            src="@/assets/instagram-square.svg" alt="facebook"
          />
          <h6 class="social-text text-center pl-2">Instagram</h6>
        </div>
      </div>
      <div class="mr-2">
        <div class="border rounded p-2 mb-1 d-flex align-items-center">
          <img class="social-icon" :class="customSocialIconStyle"
            src="@/assets/linkedin-square.svg" alt="facebook" 
          />
          <h6 class="social-text text-center pl-2">Linkedin</h6>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'socialMedia',
  props: {
    title: {
      type: String,
      required: false,
      default: "",
    },
    customSocialIconStyle: {
      type: String,
      required: false,
      default: "",
    }
  }
}
</script>

<style lang="scss" scoped>
 @import './SocialMedia.scss';
</style>

