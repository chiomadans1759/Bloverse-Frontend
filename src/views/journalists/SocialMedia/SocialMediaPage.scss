@import 'sass-rem';

#bl-social-media {
  height: 100vh;
  .main-content {
    padding: 0;
    .social-media-form {
      margin: 0;
      justify-content: center;
      .bl-form-header {
        margin-bottom: 2rem !important;
        &>h3 {
          color: #252525;
          font-weight: bold;
          text-transform: capitalize;
        }
      }
      .bl-form-btn {
        background-image: linear-gradient( to right, #003CE1, #2F80ED);
      }
      input, button {
        border-radius: rem(2px) !important;
        font-size: rem(16px);
      }
      input::placeholder {
        color: #B7B7B7;
      }
      .add-link {
        color: #096DD9;
      }
    }
  }
  .footer-container {
    position: fixed;
    bottom: 0;
    color: #B7B7B7;
    line-height: rem(40px);
  }
}
