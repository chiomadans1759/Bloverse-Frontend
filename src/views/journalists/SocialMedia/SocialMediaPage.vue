<template>
  <main id="bl-social-media" class="container-fluid p-0 bl-social-media">
    <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" customClass="back-nav"/>
    <div class="container main-content">
      <div class="row social-media-form">
        <form>
          <div class="form-group bl-form-header">
            <h3>Update Social Media</h3>
          </div>
          <div class="form-group">
            <label for="facebook">Facebook</label>
            <input type="text" class="form-control" id="facebook"
              aria-describedby="facebook" placeholder="https://facebook.com/username"
            >
          </div>
          <div class="form-group">
            <label for="twitter">Twitter</label>
            <input type="text" class="form-control" id="twitter"
              aria-describedby="twitter" placeholder="https://twitter.com/username"
            >
          </div>
          <div class="form-group">
            <label for="linkedin">LinkedIn</label>
            <input type="text" class="form-control" id="linkedin"
              aria-describedby="linkedin" placeholder="https://linkedin.com/username"
            >
          </div>
          <div class="form-group text-right pr-3 add-link">
            <span><i class="fas fa-plus mr-1"></i> Add</span>
          </div>
          <br />
          <button type="submit" class="btn btn-primary btn-block bl-form-btn">
            Save
          </button>
        </form>
      </div>
    </div>
    <div class="footer-container">
      <p><span class="bloverse">bloverse</span> 2019. All Rights Reserved</p>
    </div>
  </main>
</template>

<script>
import Navbar from '@/components/Navbar/Navbar.vue';

export default {
  name: 'SocialMedia',
  components: { Navbar },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true,
    }
  }
}
</script>

<style lang="scss" scoped>
@import "./SocialMediaPage";
</style>
