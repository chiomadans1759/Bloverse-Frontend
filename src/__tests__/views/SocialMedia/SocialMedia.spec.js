import { mount, RouterLinkStub  } from '@vue/test-utils'
import SocialMedia from '@/views/journalists/SocialMedia/SocialMediaPage.vue';

describe('SocialMedia Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(SocialMedia, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
