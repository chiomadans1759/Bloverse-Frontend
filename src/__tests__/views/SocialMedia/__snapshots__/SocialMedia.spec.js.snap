// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia Component should render correctly 1`] = `
<main
  class="container-fluid p-0 bl-social-media"
  id="bl-social-media"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5"
    >
      <a
        class="backlink"
        href="/settings"
      >
        <i
          class="fal fa-long-arrow-left fa-2x back-icon"
        />
         
        <span
          class="back-text"
        >
          Back
        </span>
      </a>
       
      <a
        class="router-link"
      >
        <img
          class="nav-logo"
          src="@/assets/Logo.svg"
        />
         
        <!---->
      </a>
       
      <!---->
       
      <!---->
    </nav>
  </header>
   
  <div
    class="container main-content"
  >
    <div
      class="row social-media-form"
    >
      <form>
        <div
          class="form-group bl-form-header"
        >
          <h3>
            Update Social Media
          </h3>
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="facebook"
          >
            Facebook
          </label>
           
          <input
            aria-describedby="facebook"
            class="form-control"
            id="facebook"
            placeholder="https://facebook.com/username"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="twitter"
          >
            Twitter
          </label>
           
          <input
            aria-describedby="twitter"
            class="form-control"
            id="twitter"
            placeholder="https://twitter.com/username"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="linkedin"
          >
            LinkedIn
          </label>
           
          <input
            aria-describedby="linkedin"
            class="form-control"
            id="linkedin"
            placeholder="https://linkedin.com/username"
            type="text"
          />
        </div>
         
        <div
          class="form-group text-right pr-3 add-link"
        >
          <span>
            <i
              class="fas fa-plus mr-1"
            />
             Add
          </span>
        </div>
         
        <br />
         
        <button
          class="btn btn-primary btn-block bl-form-btn"
          type="submit"
        >
          
          Save
        
        </button>
      </form>
    </div>
  </div>
   
  <div
    class="footer-container"
  >
    <p>
      <span
        class="bloverse"
      >
        bloverse
      </span>
       2019. All Rights Reserved
    </p>
  </div>
</main>
`;
